File: dotnet/AdventOfCode21/Day02.cs
namespace AdventOfCode21
{
    internal static class Day02
    {
        public static void Execute()
        {
			string[] completeMoves = File.ReadAllLines("day02.txt");
			int depth = 0;
			int aim = 0;
			int horizon = 0;
			for (int i = 0; i < completeMoves.Length; i++)
			{
				string[] command = completeMoves[i].Split(' ');
				int step = Convert.ToInt32(command[1]);
				switch (command[0])
				{
					case "forward":
						horizon += step;
						depth += step * aim;
						break;
					case "up":
						aim -= step;
						break;
					case "down":
						aim += step;
						break;
				}
			}
			Console.WriteLine($"Day 02.01 : depth = {depth} - horizon = {horizon}");
			Console.WriteLine($"Day 02.02 : aim = {aim} for {depth * horizon}");
		}
	}
}

File: rust/src/day02.rs
use std::fs;

pub fn execute() -> () {
    let measures = fs::read_to_string("src/day02.txt").expect("Unable to read file");
    let mut depth: i32 = 0;
    let mut aim: i32 = 0;
    let mut horizon: i32 = 0;
    for line in measures.lines() {
        let mut splitted = line.split_whitespace();
        let direction = splitted.next().unwrap();
        let step = splitted.next().unwrap().parse::<i32>().unwrap();
        match direction {
            "forward" => {
                horizon += step;
                depth += step * aim;
            },
            "up" => aim -= step,
            "down" => aim += step,
            _ => panic!(),
        }
    }
    println!("Day 02.01 : depth = {} - horizon = {}", depth, horizon);
    println!("Day 02.02 : aim {} for a total of {}", aim, depth * horizon);
}
